java/org/apache/tomcat/util/net/jsse/PEMFile.java

```diff
@@ -624,6 +626,16 @@ private byte[] fromHex(String hexString) {
             }
             return bytes;
         }
+
+
+        private String toDottedOidString(byte[] oidBytes) {
+            try {
+                Oid oid = new Oid(oidBytes);
+                return oid.toString();
+            } catch (GSSException e) {
+                throw new IllegalArgumentException(e.getMajorString());
```

Comment:
I think this only need to return a hex string of `oidBytes` if it does not follow format as the message in GSSException only show a tip 'Improperly formatted ASN.1 DER encoding for Oid' which not help.
```suggestion
                return HexUtils.toHexString(oidBytes)
```

Reply:
I am fine with either. What about:
``
throw new IllegalArgumentException(e.getMajorString() + ": " + HexUtils.toHexString(oidBytes));
```
Combining both. WDYT?

Reply:
I still think it's better to just return it, because the place where the method is called already makes it clear where there is a problem with the OID. (encryption algorithm with OID or pseudo random function with OID etc.)

Reply:
I see, right. Then it is the user's problem to investigate the file.

Reply:
Yes. ; )